docs/persistence.rst

```diff
@@ -0,0 +1,80 @@
+.. _persistence:
+
+Secure persistence with skops
+=============================
+
+.. warning::
+
+   This feature is very early in development, which means that the API is
+   unstable and that it is **not secure** for the moment. Therefore, the same
+   caution as for pickle should be applied: Don't load from sources that you
+   don't trust. In the future, more security will be added.
+
+Skops offers a way to save and load sklearn models without using :mod:`pickle`.
+Pickle is the standard serialization format for sklearn and for Python in
+general. It has the big advantage that it can be used for almost all Python code
+but this flexibility also means that it's inherently insecure. As the Python
+docs say:
+
+    The pickle module is not secure. Only unpickle data you trust. It is
+    possible to construct malicious pickle data which will execute arbitrary
+    code during unpickling. Never unpickle data that could have come from an
+    untrusted source, or that could have been tampered with.
+
+With the :func:`skops.io.save` and :func:`skops.io.load` functions, we aim at
+providing a secure method to load and save sklearn code. In contrast to pickle,
+these functions cannot be used to save arbitrary Python code, but they bypass
+pickle and are thus more secure.
+
+Usage
+-----
+
+Using :func:`skops.io.save` and :func:`skops.io.load` is quite simple. Below is
+an example:
+
+.. code:: python
+
+    from sklearn.linear_model import LogisticRegression
+    from skops.io import load, save
+
+    clf = LogisticRegression(random_state=0, solver="liblinear")
+    clf.fit(X_train, y_train)
+    save(clf, "my-logistic-regression.skops")
+    # ...
+    loaded = load("my-logistic-regression.skops")
+    loaded.predict(X_test)
+
+At the moment, we support the vast majority of sklearn estimators. This includes
+complex use cases such as :class:`sklearn.pipeline.Pipeline`,
+:class:`sklearn.model_selection.GridSearchCV`, classes using Cython code, such
+as :class:`sklearn.tree.DecisionTreeClassifier`, etc. If you discover an sklearn
+estimator that does not work, please open an issue on the skops `GitHub page
+<https://github.com/skops-dev/skops/issues>`_ and let us know.
+
+In contrast to pickle, skops cannot persist arbitrary Python code. In
+particular, this means that if you have custom functions (say, a custom function
+to be used with :class:`sklearn.preprocessing.FunctionTransformer`), it will not
+work. However, most ``numpy`` and ``scipy`` functions should work. Therefore,
+you can actually save built-in functions like``numpy.sqrt``.
```

Comment:
```suggestion
In contrast to pickle, skops cannot persist arbitrary Python code. This means 
if you have custom functions (say, a custom function to be used 
with :class:`sklearn.preprocessing.FunctionTransformer`), it will not
work. However, most ``numpy`` and ``scipy`` functions should work. Therefore,
you can actually save built-in functions like``numpy.sqrt``.
```